main/src/main/scala/sbt/Defaults.scala

```diff
@@ -1781,7 +1781,11 @@ object Defaults extends BuildCommon {
           // by the test task layers may be cached without evicting the runtime classloader layers. The
           // cache size should be a multiple of two to support snapshot layers.
           ClassLoaderCache(4)
-        )
+        ),
+        bgCopyClasspath in bgRun :=
+          (Test / classLoaderLayeringStrategy).value != ClassLoaderLayeringStrategy.ShareRuntimeDependenciesLayerWithTestDependencies,
```

Comment:
We might want to change this to

```scala
bgCopyClasspath in bgRun := {
  val old = (bgCopyClasspath in bgRun ).value
  old && (Test / classLoaderLayeringStrategy).value != ClassLoaderLayeringStrategy.ShareRuntimeDependenciesLayerWithTestDependencies  
}
```

This is because `bgCopyClasspath` is currently provided at `globalJvmCore`, which means that the build user can opt-out of it as:

```scala
ThisBuild / bgCopyClasspath := false
```

Without `old`, it is going to undo the wider scope.

Reply:
Aah, good call!

Reply:
Fixed in https://github.com/sbt/sbt/pull/4601/commits/399dd920b0578609936b0d15562672d4dbfd57f3